nixos/modules/config/system-path.nix

```diff
@@ -120,18 +120,17 @@ in
         "/share/vim-plugins"
       ];
 
-    environment.outputsToLink = [ "bin" "lib" "out" ];
-
     system.path = pkgs.buildEnv {
       name = "system-path";
       paths =
-        lib.filter (drv: drv != null && drv != (drv.dev or null))
-          (lib.concatMap (drv:
-            [ drv ] ++ map (outputName: drv.${outputName}.outPath or null) config.environment.outputsToLink)
-           config.environment.systemPackages);
-      inherit (config.environment) pathsToLink;
+        # The default output probably shouldn't be globally configurable.
+        # Services and users should specify them explicitly unless they want this default.
+        map (p: if p.outputUnspecified or false then p.bin or p.out or p else p)
```

Comment:
I'm still not certain how to best make it customizable, but the `nix-env` part really seems clear, so I implemented that now. People would better get that functionality early https://github.com/NixOS/nix/pull/815.